File: leetcode-101_symmetric-tree.js
//判断一棵二叉树是否对称
var isSymmetric = function(root) {
    return root === null || isSymmetricHelp(root.left, root.right);
}

var isSymmetricHelp = function (left, right) {
    if (left === null || right === null) {
        return left === right;
    }
    if (left.val !== right.val) {
        return false;
    }
    return isSymmetricHelp(left.left, right.right) && isSymmetricHelp(left.right, right.left);
}

File: leetcode-101_symmetric-tree.java
/**
 * 判断一个二叉树是否对称
 */
public boolean isSymmetric(TreeNode root) {
    return root == null || symmetric(root.left, root.right);
}

public boolean symmetric(TreeNode left, TreeNode right) {
    if (left == null || right == null) {
        return left == right;
    }
    if (left.val != right.val) {
        return false;
    }
    return symmetric(left.left, right.right) && symmetric(right.left, left.right);
}
